src/core/database/database.service.ts

```diff
@@ -557,16 +557,20 @@ export class DatabaseService {
 
   async deleteNodeNew<TObject extends Resource>({
     object,
-    baseNodeLabels,
     uniqueProperties = {},
   }: {
     object: TObject;
-    baseNodeLabels: string[];
     uniqueProperties?: UniqueProperties<TObject>;
   }) {
+    const nodeLabelsResult = await this.db
+      .query()
+      .matchNode('node', { id: object.id })
```

Comment:
```suggestion
      .matchNode('node', 'BaseNode', { id: object.id })
```
Should we limit this node selection to only base nodes?